### PR TITLE
feat: improved text tracks settings labels

### PR DESCRIPTION
**Pull request by wseymour15**
## Description
Based off of an audit that a Brightcove client received, we are updating the text tracks settings menu with improved labels. This change is regarding Web Content Accessibility Guidelines (WCAG). 

Screen Shot 2023-01-31 at 3 08 32 PM

## Specific Changes proposed
- Labels added/improved to fields in the settings menu
- CSS so labels are displayed next to inputs
- Translations for labels
- Updates docs regarding missing translations.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gkatsev** [comment]
Not that it shouldn't change, but the names are taken from the CVAA, which has these user settings requirements.

**Comment by OwenEdwards** [comment]
Personally, I think the text changes are fine. But my concern with the layout is that the "Font Size", "Text Edge Style", and "Font Family" controls are so close to the three "Opacity" controls that it isn't clear visually that the "Opacity" fields are each part of a fieldset with the legend "Text", "Text Background", and "Caption Area Background". That could be fixed by increasing the spacing, or adding an outline for each fieldset.

**Comment by wseymour15** [comment]
> That could be fixed by increasing the spacing, or adding an outline for each fieldset.

Good suggestion, I updated the CSS a bit to increase the spacing between some of these fields:

Screen Shot 2023-02-01 at 10 17 25 AM

**Comment by mister-ben** [comment]
It looks quite cramped at a smallish size. It might look better with color and opacity to be slightly indented at this size. Maybe text/text background in as bold heading?

grafik

File: lang/de.json [hunk 1]
```diff
@@ -86,6 +86,10 @@
   "Seek to live, currently playing live": "Zur Live-Übertragung wechseln. Es wird aktuell live abgespielt.",
   "Exit Picture-in-Picture": "Bild-im-Bild-Modus beenden",
   "Picture-in-Picture": "Bild-im-Bild-Modus",
-  "No content": "Kein Inhalt"
+  "No content": "Kein Inhalt",
+  "Color": "Farbe",
+  "Opacity": "Opazität",
+  "Text Background": "Text-Hintergrund",
```
Comment (mister-ben) [inline lang/de.json:92]:
```suggestion
  "Opacity": "Deckkraft",
  "Text Background": "Texthintergrund",
```

File: lang/en.json [hunk 2]
```diff
@@ -87,5 +87,9 @@
   "{1} is loading.": "{1} is loading.",
   "Exit Picture-in-Picture": "Exit Picture-in-Picture",
   "Picture-in-Picture": "Picture-in-Picture",
-  "No content": "No content"
+  "No content": "No content",
+  "Color": "Color",
```
Comment (mister-ben) [inline lang/en.json:91]: We're going to need an en-GB :)

Comment (wseymour15) [reply lang/en.json:91]: Good call, created this issue here: https://github.com/videojs/video.js/issues/8105

**Comment by wseymour15** [comment]
> It looks quite cramped at a smallish size. It might look better with color and opacity to be slightly indented at this size. Maybe text/text background in as bold heading?
You're right, that does not look as expected, and nor did a lot of mobile devices when the menu was opened full screen. I updated this with bolded headers and a few small styling changes to make it look better in most cases. The only flaw I found is that in some medium sized players, the user is now forced to scroll to get to the bottom of the menu. If this is an issue let me know, but I believe in most cases this will look much better.
Screen Shot 2023-02-02 at 1 14 20 PM

File: lang/en.json (hunk 2 above)
Comment (gkatsev) [reply lang/en.json:91]: I think mastodon using british naming balances things out 😉

**Comment by gkatsev** [comment]
I think it's fine for now, even with the scrolling. I built it out 10 years and 2 months ago, and it needs to be re-designed from scratch at this point.

**Comment by roman-bc-dev** [comment]
I agree with @mister-ben that a small indent to the left of the labeled settings (color, opacity) would also improve at-a-glance legibility. Overall, it looks much cleaner!

**Comment by wseymour15** [comment]
> I agree with @mister-ben that a small indent to the left of the labeled settings (color, opacity) would also improve at-a-glance legibility. Overall, it looks much cleaner!

@roman-bc-dev This was more work then I expected 😅 One of these label/select elements was wrapped in a span, the other was not.. so I updated the HTML to ensure that both are wrapped in a span element, updated the css, and also a few test fixes :)

**Review by roman-bc-dev — APPROVED** [review]

**Pull request merged**